swift/extractor/SwiftExtractor.cpp

```diff
@@ -53,12 +54,16 @@ static std::string getFilename(swift::ModuleDecl& module, swift::SourceFile* pri
   if (primaryFile) {
     return primaryFile->getFilename().str();
   }
-  // Several modules with different name might come from .pcm (clang module) files
-  // In this case we want to differentiate them
-  std::string filename = module.getModuleFilename().str();
-  filename += "-";
-  filename += module.getName().str();
-  return filename;
+  // PCM clang module
+  if (module.isNonSwiftModule()) {
+    // Several modules with different name might come from .pcm (clang module) files
+    // In this case we want to differentiate them
```

Comment:
this does not explain why we strip the directory. We should probably say `pcm` files come from caches that might be located in different places, but are unambigouously identified by the base filename

Reply:
I need to rebase https://github.com/github/codeql/pull/9922 on this one any way, so we can maybe just merge this as it is and I'll update the comment in mine